### DISCUSSION TITLE
Message or key does not conform to a valid OpenPGP format

### DISCUSSION THREAD

**Discussion by sasa130** [openpgpjs/openpgpjs#1626] [Q&A]
Hi, I am getting an error when calling readMessage function. 
  const message = await readMessage({
        binaryMessage: encryptedData
  })

where encryptedData is an encrypted zip file from S3.  The specific error is:

Error during parsing. This message / key probably does not conform to a valid OpenPGP format.

The scenario is that a third party is encrypting the file before sending it to another system which decrypts it.
The encryption is done by a lambda written in Python using pgpy library and it is being decrypted by another lambda using Typescript with openpgp.  This model uses asymmetric algorithm and the key pair is generated by another typecript lambda (using openpgp).

We are sure that the zip file is encrypted and decrypted with the corresponding public key of the private key.  Can someone explain what the error is about? 

We have performed the following test scenarios to check where the problem could be (all of them using the same generated public and private key pair):
1.  Python component is able to encrypt and decrypt zip file successfully 
2. Typescript component is able to encrypt and decrypt zip file successfully

The weird thing is running the **gpg decrypt** command also works against the encrypted file produced by the Python component, but fails on the **readMessage** function above.  At this point, my guess is there are some default values that aren't aligned with the 2 libraries but I've tried passing various config options on the readMessage function but it always throws the same error for files encrypted using pgpy.

Can someone help please?

**Comment by larabr** [comment]
Are you sure the file is in binary format, as opposed to armored? Other than that, we'd need to inspect the encrypted file to help you further.

**Reply by sasa130** [reply]
Yes, it is in binary format, I've attached the sample encrypted zip file.  The relevant Python code snippets are below

```
 if outFormat == "ASCII":
        encryptFile(targetFile, publicKey)
 elif outFormat == "BINARY":
        encryptBinaryFile(targetfile, `publicKey)`
```

```
def encryptBinaryFile(path, key):
    unencryptedMessage = pgpy.PGPMessage.new(path, file=True)
    encryptedMessage = key.encrypt(unencryptedMessage)
    encryptedFile = open(path, "wb")
    encryptedFile.write(bytes(encryptedMessage))
```
[test.zip](https://github.com/openpgpjs/openpgpjs/files/11129262/test.zip)

**Reply by larabr** [reply]
The data you provided is in armored format (`-----BEGIN PGP MESSAGE-----...`), you need to pass the `armoredMessage` option to `openpgp.readMessage` to read it.

**Reply by sasa130** [reply]
Thanks @larabr  for the response.  It worked when I changed to armoredMessage instead of binaryMessage but also I had to specific 'utf-8' parameter when calling createReadStream on the encrypted file stored locally.  However, when I tested the actual scenario, the encrypted file is downloaded from S3 and the readMessage function throws the error below:

' Misformed armored text'

I'm confused though how it ended up having to use armoredMessage.  Is it more to do with the Python code encrypting the file or some S3 bucket settings?  If I encrypt using openpgp, using binaryMessage for the readMessage function works just fine.  I've already looked up pgpy library and it has simple encrypt function which doesn't have any other config parameters.

**Reply by larabr** [reply]
Whether the encrypted output is binary or armored is purely a function of the encrypting library -- openpgp data can be converted into either formats. Make sure you're reading the stream properly from S3, but you should know in advance whether it's binary or armored.

**Reply by sasa130** [reply]
It was indeed because the encrypted file wasn't in binary format.  Thank you so much @larabr !
